app/dose-event-service/dose-event.service.ts

```diff
@@ -1,54 +1,124 @@
 import {Injectable} from "@angular/core";
 import {DoseEvent} from "../dose-event/dose-event";
-import {DOSE_EVENTS} from "./dose-event-mocks";
-import {DOSE_EVENTS2} from "./dose-event-mocks-two";
-import {DOSE_EVENTS3} from "./dose-event-mocks-three";
+import {BACKEND_URL} from "../backend-config/backend-config";
+import {AuthHttp} from "../angular2-jwt";
+import {Headers, RequestOptions} from "@angular/http";
 
 /**
  * Service for fetching and saving DoseEvent instances.
  */
 @Injectable()
 export class DoseEventService {
 
+    private doseEventsUrl: string = `${BACKEND_URL}/dose-events`;
+
+    constructor(
+        private http: AuthHttp
+    ) {
+    }
+
+    /**
+     * Saves all provided doseEvents to the server.
+     * @param doseEvents to save.
+     * @returns {Promise<Array<DoseEvent>>}.
+     */
+    public saveList(
+        doseEvents: Array<DoseEvent>
+    ): Promise<Array<DoseEvent>> {
+
+        const headers = new Headers(
+            {
+                "Content-Type": "application/json"
+            }
+        );
+
+        const options = new RequestOptions(
+            {
+                headers: headers
+            }
+        );
+
+        return this.http.put(
+            this.doseEventsUrl,
+            JSON.stringify(doseEvents),
+            options
+        ).toPromise().then(
+            res => res.json()
+        ).catch(
+            this.handleError
+        );
+    }
+
     /**
      * Get a list of DoseEvents.
-     * @returns {Promise<Array<DoseEvent>>}
+     * @returns {Promise<Array<DoseEvent>>}.
      */
     public list(): Promise<Array<DoseEvent>> {
-        return Promise.resolve(
-            DOSE_EVENTS
+        return this.loadPage(
+            new Date(),
+            "next"
+        );
+    }
+
+    /**
+     * Loads a single page of events, which corresponds to a specific time.
+     * @param startAt - time currently viewing.
+     * @param dir - direction; either "next" or "prev".
+     * @returns {Promise<Array<DoseEvent>>}.
+     */
+    public loadPage(
+        startAt: Date,
+        dir: string
+    ): Promise<Array<DoseEvent>> {
+        return this.http.get(
+            `${this.doseEventsUrl}?startAt=${startAt.getTime()}&dir=${dir}`
+        ).toPromise().then(
+            (response) => response.json()
+        ).catch(
+            this.handleError
+        );
+    }
```

Comment:
AuthHttp was ignoring my attempts at providing the query params via other means, so I just made them part of the URL.